feat: allow parserFn to be overridden

JaspervanRiet opened the pull request:
This PR allows the `parserFn` function to be overridden.

We have the following use case: some of our GraphQL calls are quite heavy and cause slowdown on the UI thread when parsing the JSON. A simple fix to this is to parse this JSON in an isolate. Currently this isn't convenient using this library. We have to copy the generated query classes and explicitly remove the line that sets the `parserFn` function. We then manually call the exact same parsing functionality from our isolate, which solves the slowdown. This PR solves this issue, without breaking current functionality.

I did not add tests to this PR. Happy to add them, but will need a few pointers on the test architecture in this repo.

Example
=====

Before:

```dart
  final result = await client.query(_MyQuery(
        [...]
      ));
  // This does an implicit JSON parse that is too heavy for us!

  return Something.fromGraphQl(result.parsedData);
```

after:

```dart
  final json = await client.query(_MyQuery(
        [...],
        parserFn: null,
      ));

  return await compute(json)
```

budde377 reviewed (changes requested):
Thanks for submitting your PR. You've only changed the test, so this does nothing but breaking the tests.

JaspervanRiet commented:
Silly me for doing a simple CTRL F and not really looking at the paths. I dived into the printer and made the changes there. Locally it seems to be going the right way, mind taking another look? If not, would appreciate a pointer on where to proceed.

budde377 commented:
Can you try and elaborate a bit on how this will solve your use case? Communication between isolates are asynchronous, but these functions are not?

JaspervanRiet commented:
You're completely right. Sorry, we're in a bit of a rush to launch in production. Added the use case to the description and added a new commit that _actually_ implements what we want, which is to be able to explicitly set parserFn to null, so that we get JSON back.

budde377 commented:
No worries. I think you might be able to just call `response.data` as opposed to `response.parsedData`. The parsed data is computed lazily so this should solve your problem?

JaspervanRiet commented:
Yep that works! Thanks, I'll get out of your hair :)
